pkg/platform/update-api/update_api_test.go

```diff
@@ -0,0 +1,149 @@
+			Expected: UpdateStatus{
+				UpdateState:      Staged,
+				AvailableUpdates: []string{"0.4.0", "0.3.4", "0.3.3", "0.3.2", "0.3.1", "0.3.0"},
+				ChosenUpdate: &UpdateImage{
+					Arch:    "x86_64",
+					Version: "0.4.0",
+					Variant: "aws-k8s-1.15",
+				},
+				ActivePartition: &StagedImage{
+					Image: UpdateImage{
+						Arch:    "x86_64",
+						Version: "0.3.4",
+						Variant: "aws-k8s-1.15",
+					},
+					NextToBoot: true,
+				},
+				StagingPartition: &StagedImage{
+					Image: UpdateImage{
+						Arch:    "x86_64",
+						Version: "0.4.0",
+						Variant: "aws-k8s-1.15",
+					},
+					NextToBoot: false,
+				},
+				MostRecentCommand: &CommandResult{
+					CmdType:    prepare,
+					CmdStatus:  Success,
+					Timestamp:  "2020-07-10T06:44:58.766493367Z",
+					ExitStatus: new(int32),
+					Stderr:     &update_string,
+				},
+			},
+		},
+		{
+			Name:             "Update ready",
+			UpdateStatusJson: []byte(`{"update_state":"Ready","available_updates":["0.4.0","0.3.4","0.3.3","0.3.2","0.3.1","0.3.0"],"chosen_update":{"arch":"x86_64","version":"0.4.0","variant":"aws-k8s-1.15"},"active_partition":{"image":{"arch":"x86_64","version":"0.3.4","variant":"aws-k8s-1.15"},"next_to_boot":false},"staging_partition":{"image":{"arch":"x86_64","version":"0.4.0","variant":"aws-k8s-1.15"},"next_to_boot":true},"most_recent_command":{"cmd_type":"activate","cmd_status":"Success","timestamp":"2020-07-10T06:47:19.903337270Z","exit_status":0,"stderr":""}}`),
+			Expected: UpdateStatus{
+				UpdateState:      Ready,
+				AvailableUpdates: []string{"0.4.0", "0.3.4", "0.3.3", "0.3.2", "0.3.1", "0.3.0"},
+				ChosenUpdate: &UpdateImage{
+					Arch:    "x86_64",
+					Version: "0.4.0",
+					Variant: "aws-k8s-1.15",
+				},
+				ActivePartition: &StagedImage{
+					Image: UpdateImage{
+						Arch:    "x86_64",
+						Version: "0.3.4",
+						Variant: "aws-k8s-1.15",
+					},
+					NextToBoot: false,
+				},
+				StagingPartition: &StagedImage{
+					Image: UpdateImage{
+						Arch:    "x86_64",
+						Version: "0.4.0",
+						Variant: "aws-k8s-1.15",
+					},
+					NextToBoot: true,
+				},
+				MostRecentCommand: &CommandResult{
+					CmdType:    activate,
+					CmdStatus:  Success,
+					Timestamp:  "2020-07-10T06:47:19.903337270Z",
+					ExitStatus: new(int32),
+					Stderr:     new(string),
+				},
+			},
+		},
+	}
+	for _, tc := range cases {
+		var unmarshaledStatus UpdateStatus
+		err := json.Unmarshal(tc.UpdateStatusJson, &unmarshaledStatus)
+		assert.NoError(t, err, "failed to unmarshal into update status")
+		assert.Equal(t, tc.Expected, unmarshaledStatus)
```

Comment:
This can be made into a test table, which is handy to distinguish each as its own test:

```suggestion
		t.Run(tc.Name, func(t *testing.T) {
			var unmarshaledStatus UpdateStatus
			err := json.Unmarshal(tc.UpdateStatusJson, &unmarshaledStatus)
			assert.NoError(t, err, "failed to unmarshal into update status")
			assert.Equal(t, tc.Expected, unmarshaledStatus)
		})
```